pkg/ccl/multitenantccl/tenantcostclient/tenant_side.go

```diff
@@ -471,6 +475,25 @@ func (c *tenantSideCostController) onTick(ctx context.Context, newTime time.Time
 		c.run.fallbackRateStart = time.Time{}
 	}
 
+	// Report consumption metrics. Update local data first before sending a
+	// token bucket request to the KV servers.
+	deltaConsumption := c.run.consumption
+	deltaConsumption.Sub(&c.run.lastExportedConsumption)
+	c.metrics.TotalRU.Inc(deltaConsumption.RU)
+	c.metrics.TotalKVRU.Inc(deltaConsumption.KVRU)
+	c.metrics.TotalReadBatches.Inc(int64(deltaConsumption.ReadBatches))
+	c.metrics.TotalReadRequests.Inc(int64(deltaConsumption.ReadRequests))
+	c.metrics.TotalReadBytes.Inc(int64(deltaConsumption.ReadBytes))
+	c.metrics.TotalWriteBatches.Inc(int64(deltaConsumption.WriteBatches))
+	c.metrics.TotalWriteRequests.Inc(int64(deltaConsumption.WriteRequests))
+	c.metrics.TotalWriteBytes.Inc(int64(deltaConsumption.WriteBytes))
+	c.metrics.TotalSQLPodsCPUSeconds.Inc(deltaConsumption.SQLPodsCPUSeconds)
+	c.metrics.TotalPGWireEgressBytes.Inc(int64(deltaConsumption.PGWireEgressBytes))
+	c.metrics.TotalExternalIOEgressBytes.Inc(int64(deltaConsumption.ExternalIOEgressBytes))
+	c.metrics.TotalExternalIOIngressBytes.Inc(int64(deltaConsumption.ExternalIOIngressBytes))
+	c.metrics.TotalCrossRegionNetworkRU.Inc(deltaConsumption.CrossRegionNetworkRU)
+	c.run.lastExportedConsumption = c.run.consumption
+
 	// Should a token bucket request be sent? It might be for a retry or for
 	// periodic consumption reporting.
 	if c.run.shouldSendRequest || c.shouldReportConsumption() {
```

Comment:
sendTokenBucketRequest is also creating a consumption delta and reporting that delta to the server. What do you think of moving  the metric increment into sendTokenBucketRequest so that we only collect and track the deltas once?